tools/c7n_azure/c7n_azure/session.py

```diff
@@ -228,6 +228,8 @@ def client(self, client, vault_url=None):
             # 2020-06-01 is not supported, but 2019-11-01 is working as expected.
             if client == 'azure.mgmt.costmanagement.CostManagementClient':
                 client_args['raw_request_hook'] = cost_query_override_api_version
+            elif client == 'azure.mgmt.eventhub.EventHubManagementClient':
```

Comment:
I would suggest update the api version on event hub resource to 2021 variant instead of hacking into session.

Reply:
Session hacking is removed. 
I updated `azure-mgmt-eventhub` to `11.0.0` in that PR - [9197](https://github.com/cloud-custodian/cloud-custodian/pull/9117). The default api version for the new package version is 2021-11-01. Hope this's right